src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -3663,6 +3645,9 @@ internal void SetToConnected()
             // some point in time update the perf counter as well.
             _isConnected = true;
             _isDisconnected = false;
+            _pollPendingConnect = false;
```

Comment:
There are lots of callers where this won't be true, right?

I think it's slightly better to clear this flag in the caller before calling this, then assert here that it's false.